tests/unit/development/test_base_component.py

```diff
@@ -473,3 +474,37 @@ def test_debc027_component_error_message():
         + "keyword argument: `asdf`\n"
         + "Allowed arguments: {}".format(", ".join(sorted(html.Div()._prop_names)))
     )
+
+
+def test_set_random_id():
+    app = Dash(__name__)
+
+    input1 = dcc.Input(value="Hello")
+    input2 = dcc.Input(value="Hello")
+    output1 = html.Div()
+    output2 = html.Div()
+    output3 = html.Div(id="output-3")
+
+    app.layout = html.Div([input1, input2, output1, output2, output3])
+
+    @app.callback(Output(output1, "children"), Input(input1, "value"))
+    def update(v):
+        return f"Input 1 {v}"
+
+    @app.callback(Output(output2, "children"), Input(input2, "value"))
+    def update(v):
+        return f"Input 2 {v}"
+
+    @app.callback(
+        Output("output-3", "children"), Input(input1, "value"), Input(input2, "value")
+    )
+    def update(v1, v2):
+        return f"Output 3 - Input 1: {v1}, Input 2: {v2}"
+
+    # Verify the auto-generated IDs are stable
+    assert output1.id == "e3e70682-c209-4cac-629f-6fbed82c07cd"
+    assert input1.id == "82e2e662-f728-b4fa-4248-5e3a0a5d2f34"
+    assert output2.id == "d4713d60-c8a7-0639-eb11-67b367a9c378"
+    assert input2.id == "23a7711a-8133-2876-37eb-dcd9e87a1613"
+    # we make sure that the if the id is set explicitly, then it is not replaced by random id
+    assert output3.id == "output-3"
```

Comment:
Could we do a quick assertion to verify that the callbacks are correctly fired? You can either assert that that the values are there by looking up the element in the DOM (see other tests) or you can take a screenshot.

Reply:
13e008f0